Bot.Core/BotSystemRunner.cs

```diff
@@ -27,6 +28,7 @@ public BotSystemRunner(IServiceProvider parentServices, IBotSystem system)
 
             m_serviceProvider.GetService<IVersionProvider>().InitializeVersions();
 
+            m_ghostTownCleanup = new GhostTownCleanup(m_serviceProvider);
```

Comment:
I would actually probably construct this as a service in `ServiceFactory.RegisterBotServices` (which is the thing currently responsible for the construction of most of the top-level things dealing with bots).